c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1183,6 +1180,8 @@ class DeviceCachingAllocator {
   // device statistics
   DeviceStats stats;
 
+  c10::DeviceIndex device_id;
```

Comment:
separate changes related to adding `device_id` as a new member from changes related to per_process_memory_fraction, for ease of reviewing